authors/apps/profiles/test/test_user_profile.py

```diff
@@ -0,0 +1,117 @@
+import json
+from django.urls import reverse
+from rest_framework.views import status
+from rest_framework.test import APITestCase, APIClient
+
+
+class LoginTestCase(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.login_url = reverse('authentication:auth-login')
+        self.signup_url = reverse('authentication:auth-register')
+        self.data_john = {
+            "user": {
+                "username": "johndoe",
+                "email": "john@gmail.com",
+                "password": "@Qwerty12345"
+            }}
+        self.data_jane = {
+            "user": {
+                "username": "janedoe",
+                "email": "jane@gmail.com",
+                "password": "@Qwerty12345"
+            }}
+        self.profile_data = {
+            "first_name": "kwame",
+            "last_name": "asiago",
+            "gender": "F",
+            "bio": "my bio"
+        }
+
+        self.invalid_gender = {
+            "first_name": "kwame",
+            "last_name": "asiago",
+            "gender": "G",
+            "bio": "my bio"
+        }
+
+    def get_token(self):
+        register = self.client.post(
+            self.signup_url, self.data_john, format='json')
+        self.assertEqual(register.status_code, status.HTTP_201_CREATED)
+        return register.data['token']
+
+    def get_token_jane(self):
```

Comment:
This is same with the function above. Modify the `get_token` function to take an argument(signup details) and return the token based on the details provided.